polygon/heimdall/entity.go

```diff
@@ -0,0 +1,5 @@
+package heimdall
+
+type entity interface {
```

Comment:
you would not need this if you properly implement `syncEntity` as a generic function - it will be much tidier in my opinion

Reply:
I agree that this is odd as it is. In the next PR I was going to turn this into a normal interface, and implement it inside Checkpoint/Milestone/Span. It will simplify the code there, because I need a common way to get Id and a "range" (Start/End) of each entity. So "entity" has Id, Start/End and is JSON-encodable.